<launch>	
	        <!--  ************** Navigation ***************  -->
<arg name="move_forward_only" default="false"/>
<arg name="node_start_delay" default="1.0" />  
  <arg name="map_file" default="/home/cleaning_robot/cleaning_ws/src/odom/map/map_p41.yaml"/>
<!-- <include file="$(find odom)/launch/odom_bringup_tf.launch" /> -->
  <!-- Map server -->
  <node pkg="map_server" name="map_server" type="map_server" args="$(arg map_file) "/>
  <include file="$(find gmcl)/examples/gmcl_diff.launch" />
	<node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen" launch-prefix="bash -c 'sleep $(arg node_start_delay); $0 $@'">
  	  	<rosparam file="$(find odom)/cfg/costmap_common_params.yaml" command="load" ns="global_costmap" />
  	 	<rosparam file="$(find odom)/cfg/costmap_common_params.yaml" command="load" ns="local_costmap" />
  		<rosparam file="$(find odom)/cfg/local_costmap_params.yaml" command="load" />
  		<rosparam file="$(find odom)/cfg/global_costmap_params.yaml" command="load" />
  		<rosparam file="$(find odom)/cfg/base_local_planner_params.yaml" command="load" />
		<param name="base_local_planner" value="dwa_local_planner/DWAPlannerROS" />
		<!-- base_local_planner::TrajectoryPlanner -->
		<!-- <param name="base_global_planner" value="global_planner/GlobalPlanner" /> -->
		<param name="base_global_planner" value="navfn/NavfnROS" />
		<!-- <param name="base_local_planner" value="teb_local_planner/TebLocalPlannerROS" /> -->
		<!-- <param name="base_local_planner" value="base_local_planner/TrajectoryPlannerROS"/> -->
		<param name="controller_frequency" value="10.0" />
		<!-- <param name="controller_patience" value="15.0" /> -->
		<!-- <param name="planner_patience" value="10.0" /> -->
		<param name="planner_frequency" value="0.0" />
		<param name="shutdown_costmaps" value="false" />
	
		<param name="clearing_rotation_allowed" value="true" />
		<!-- <param name="move_forward_only" value="$(arg move_forward_only)"/> -->
	</node>
	<!-- <node name="rviz" pkg="rviz" type="rviz" args="-d $(find odom)/cfg/rviz_dwa.rviz" required="true" /> -->

</launch>
